### PR TITLE
[db-sync] Sync `d_b_volume_snapshot` table

### PR DESCRIPTION
**Pull request by andrew-farries**
## Description

Make the `db_volume_snapshot` table able to be synced between clusters.

* Add a migration so that the table has `_lastModified` and `deleted` fields.

Records are currently hard-deleted from this table, so the `deleted` field will not be used in practice:

https://github.com/gitpod-io/gitpod/blob/99e86333318deb096044e0f25d6000293e46f26d/components/gitpod-db/src/typeorm/workspace-db-impl.ts#L776-L779

## Related Issue(s)
Part of https://github.com/gitpod-io/gitpod/issues/9198

## Release Notes
```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

File: components/gitpod-db/src/tables.ts
Comment (easyCZ) [inline components/gitpod-db/src/tables.ts:50]: Feels a bit risky to include this in the same PR as the migration. Are you happy with the risks involved in case the migration doesn't go through but db-sync gets rolled out?

**Review by easyCZ — APPROVED** [review]
LGTM

/hold for Q on risk

**Comment by geropl** [comment]
@andrew-farries Did you engage with @sagor999 before doing this? We definitely need to do this before deploying this.

/hold

**Comment by andrew-farries** [comment]
> @andrew-farries Did you engage with @sagor999 before doing this? 

I've requested a review from @sagor999.

File: components/gitpod-db/src/tables.ts
Comment (andrew-farries) [reply components/gitpod-db/src/tables.ts:50]: Good point - I'll leave the migration + config change in this PR for ease of review but I can split it into two before merging.

**Review by sagor999 — APPROVED** [review]
LGTM
(I was under impression that this table already should have been db-synced but I guess it was not? So good catch! :pray: )

File: components/gitpod-db/src/tables.ts
Comment (andrew-farries) [reply components/gitpod-db/src/tables.ts]: I've removed the `db-sync` config change from this PR and I'll add it in a follow up.

**Pull request merged**
